logrotate.c

```diff
@@ -2640,22 +2673,6 @@ static int readState(const char *stateFilename)
         }
     }
 
-    if (!debug && (f_stat.st_size == 0)) {
-        /* create the file before continuing to ensure we have write
-           access to the file */
```

Comment:
I am afraid that the comment you removed had been there for a reason.  Should we trigger any rotation if we know that we will not be able to write that state file afterwards?

Reply:
This might be a valid concern, but currently we rotate regardless of having write access or not
https://github.com/logrotate/logrotate/blob/44a097a227eb28252424fe0f2281b4e97d06169d/logrotate.c#L2940

Reply:
Indeed.  Sorry for missing it!